File: week3/Ariling/prgs-셔틀버스/main.js
function solution(n, t, m, timetable) {
  var answer = "";
  const timeArr = [];
  const busArr = [];
  // 먼저 각 타임 테이블 애들을 숫자로 바꿔준다.
  for (let i = 0; i < timetable.length; i++) {
    let hour = parseInt(timetable[i].slice(0, 3));
    let minute = parseInt(timetable[i].slice(3));
    timeArr.push(hour * 60 + minute);
  }
  timeArr.sort((a, b) => a - b);
  // 버스 시간 배열 만들기
  for (let i = 0; i < n; i++) {
    busArr.push(9 * 60 + t * i);
  }
  let idx = 0;
  // 콘의 탑승 시간 알아내기 각각 버스의 경우를 보면서
  for (let i = 0; i < busArr.length; i++) {
    let bus = busArr[i];
    let cnt = 0;
    while (idx < timeArr.length && timeArr[idx] <= bus && cnt < m) {
      idx += 1;
      cnt += 1;
    }
    if (cnt < m) {
      answer = bus;
    } else {
      // 0번부터 들어간 것이므로 idx-1을 해주고 1분 전에 얌치기를 해야 콘이 들어갈 수 있다.
      answer = timeArr[idx - 1] - 1;
    }
  }
  // 다시 문자열로 치환하기
  let hour = String(Math.floor(answer / 60));
  let minute = String(answer - Math.floor(answer / 60) * 60);
  hour = hour.length > 1 ? hour : hour.padStart(2, 0);
  minute = minute.length > 1 ? minute : minute.padStart(2, 0);
  answer = hour + ":" + minute;
  return answer;
}

File: week3/Ariling/prgs-셔틀버스/main.py
def solution(n, t, m, timetable):
    # 그래... 규칙은 알아냈다.
    answer = ''
    # 다 분으로 변환하기
    timeArr = []
    for time in timetable:
        hour = int(time[:2])*60
        minute = int(time[3:])
        timeArr.append(hour+minute)
    timeArr = sorted(timeArr)
    # 버스 시간 만들기
    busTime = [9*60+t*i for i in range(n)]
    i = 0 # 버스에 탈 크루의 인덱스
    for bus in busTime:
        cnt = 0
        while cnt < m and i < len(timeArr) and timeArr[i] <= bus:
            i += 1
            cnt += 1
        if cnt < m:
            answer = bus
        else:
            answer = timeArr[i-1] - 1 # 맨 마지막 크루보다 1분전에 해서 들어가면 된다.
    # 숫자를 문자열로 바꾸기
    hour = answer // 60
    minute = answer - (hour * 60)
    # zfill이라는 메서드로 숫자앞에 0을 붙이는 걸 알았습니당
    answer = str(hour).zfill(2) + ":" + str(minute).zfill(2)
    return answer